snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/ModuleRequest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.snomed.datastore.request;
+
+import java.util.function.Function;
+
+import com.b2international.snowowl.core.domain.TransactionContext;
+import com.b2international.snowowl.core.events.DelegatingRequest;
+import com.b2international.snowowl.core.events.Request;
+import com.b2international.snowowl.snomed.Component;
+
+/**
+ * @since 6.17
+ */
+public final class ModuleRequest<R> extends DelegatingRequest<TransactionContext, TransactionContext, R> {
+
+	public interface ModuleIdFunction extends Function<Component, String> { }
```

Comment:
Wouldn't it be better to name this type `ModuleIdProvider`?

Reply:
To keep backwards compatibility, one of the implementations of this function uses the `Component` parameter, while the other doesn't. It was originally a `Supplier` 😄